### ISSUE TITLE
Update build status of build that received SIGTERM

### ISSUE THREAD

**Issue by sugarnaoming** [screwdriver-cd/screwdriver#2754]
**What happened**:
If a SIGTERM is sent for a running build pod, the build status in the UI may not be updated properly.
If the build status is not updated successfully, the build status will remain RUNNING on the UI.
The user must press the "stop build" button on the UI screen to correct this.
> We are considering implementing K8s SoftEvict feature as a mechanism to achieve the Disk space limitation described in this issue: https://github.com/screwdriver-cd/screwdriver/issues/2722.

For example, the build status on the UI will not be updated in the following cases
- When a running build pod is deleted by `kubectl delete pod`
- When a running build pod is deleted using k8s SoftEviction function.

The reason why the build status is not updated correctly is that the launcher itself [kills the `sleep` process running in `trap`](https://github.com/screwdriver-cd/launcher/blob/master/executor/executor.go#L405) in `run.sh` before the launcher sends the [build status update request to the API](https://github.com/screwdriver-cd/launcher/blob/master/launch.go#L193).

**What you expected to happen**:
I want to update the build status on the UI correctly even if I use `kubectl delete pod` or k8s SoftEvict function.

**How to reproduce it**:
Basically, I can reproduce this by deleting the running build pod with the `kubectl delete pod` command.
> Please refer to the official documentation for more details on how to set up the SoftEviction feature.

**Comment by jithine** [comment]
@pritamstyz4ever @tkyi didn't we put some fixes in place to handle build status and pod termination ?
